pkg/kopialib/storage/s3.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2024 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"context"
+
+	"github.com/kanisterio/kanister/pkg/kopialib"
+	"github.com/kanisterio/kanister/pkg/utils"
+	"github.com/kopia/kopia/repo/blob"
+	"github.com/kopia/kopia/repo/blob/s3"
+)
+
+var _ Storage = &s3Storage{}
+
+var requiredS3Arguments = []string{
+	kopialib.BucketKey,
+}
+
+type s3Storage struct {
+	Options *s3.Options
+	Create  bool
+}
+
+func (s *s3Storage) Connect() (blob.Storage, error) {
+	return s3.New(context.Background(), s.Options, s.Create)
+}
+
+func (s *s3Storage) WithOptions(opts s3.Options) {
+	s.Options = &opts
+}
+
+func (s *s3Storage) WithCreate(create bool) {
+	s.Create = create
+}
+
+func (s *s3Storage) SetOptions(ctx context.Context, options map[string]string) error {
```

Comment:
do we really need `WithOptions` and `SetOptions` separately? it looks like both of them are just trying to set `s.Options`.

Reply:
Yes, SetOptions is a generic way to set configuration parameters for all the storage providers and its part of interface. WithOptions is not part of the interface and is a utililty which takes input specific to a provider. Its just a utility function that can be used apart from the generic method

Reply:
Removed WithOptions for now, will add later if required